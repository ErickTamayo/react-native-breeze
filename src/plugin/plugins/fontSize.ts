import { validate } from "../utils/validation";
import { PluginFunction, PluginPattern } from "./types";

export type PluginGroups = {
  key: string;
};

export const pattern: PluginPattern = ({ keys }) => {
  return new RegExp(`^text-(?<key>${keys("fontSize")})$`);
};

export const plugin: PluginFunction<PluginGroups> = ({
  input,
  groups,
  theme,
}) => {
  const { key } = groups;
  const value = theme<number>(["fontSize", key]);

  if (!validate(input, value, ["number"])) return {};

  return { fontSize: value };
};
