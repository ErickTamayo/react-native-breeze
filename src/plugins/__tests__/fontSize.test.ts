import {
  generateInput,
  shouldEvaluateTheCorrectPatternTest,
  shouldMatchOutputSnapshot,
} from "../../helpers/tests";
import { pattern, plugin } from "../fontSize";

describe("fontSize", () => {
  // prettier-ignore
  const shouldMatch = generateInput("text", ["xs", "sm", "base", "lg", "xl", "2xl", "3xl", "4xl", "5xl", "6xl"]);

  shouldEvaluateTheCorrectPatternTest(pattern, {
    shouldMatch,
    shouldNotMatch: ["unknown", "text-unknown"],
  });

  shouldMatchOutputSnapshot(pattern, plugin, shouldMatch);
});
