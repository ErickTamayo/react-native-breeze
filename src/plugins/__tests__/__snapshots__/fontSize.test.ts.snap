// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fontSize Should parse (text-2xl) correctly 1`] = `
Object {
  "fontSize": 24,
}
`;

exports[`fontSize Should parse (text-3xl) correctly 1`] = `
Object {
  "fontSize": 30,
}
`;

exports[`fontSize Should parse (text-4xl) correctly 1`] = `
Object {
  "fontSize": 36,
}
`;

exports[`fontSize Should parse (text-5xl) correctly 1`] = `
Object {
  "fontSize": 48,
}
`;

exports[`fontSize Should parse (text-6xl) correctly 1`] = `
Object {
  "fontSize": 64,
}
`;

exports[`fontSize Should parse (text-base) correctly 1`] = `
Object {
  "fontSize": 16,
}
`;

exports[`fontSize Should parse (text-lg) correctly 1`] = `
Object {
  "fontSize": 18,
}
`;

exports[`fontSize Should parse (text-sm) correctly 1`] = `
Object {
  "fontSize": 14,
}
`;

exports[`fontSize Should parse (text-xl) correctly 1`] = `
Object {
  "fontSize": 20,
}
`;

exports[`fontSize Should parse (text-xs) correctly 1`] = `
Object {
  "fontSize": 12,
}
`;
